docs/src/design/programs.md

```diff
@@ -26,6 +26,32 @@ After the body of the loop is executed, the VM checks the top of the stack again
 
 A *loop* block must always have one child, and thus, cannot be a leaf node in the tree.
 
+### Call block
+
+A **call** block is used to describe a function call which is executed in a [user context](../user_docs/assembly/execution_contexts.md). When the VM encounters a *call* block, it creates a new user context, then executes a program which hashes to the target specified by the *call* block in the new context. Thus, in order to execute a *call* block, the VM must be aware of a program with the specified hash. Otherwise, the execution fails. At the end of the *call* block, execution returns to the previous context. 
+
+
+When executing a *call* block, the VM does the following:
+1. Sets the depth of the stack to 16.
+2. Upon return, checks that the depth of the stack is 16. If so, the original stack depth is restored. Otherwise, an error occurs.
```

Comment:
Here we also check if we are in a `syscall`. Executing a `call` block inside a `syscall` also results in an error.